### ISSUE TITLE
[BUG] Rancher packaged version of Harvester UI (1.0.1) breaks VM creation in registered Harvester 1.0.0 clusters

### ISSUE THREAD

**Issue by ebauman** [harvester/harvester#2128]
**Describe the bug**
The packaging of an experimental version of the Harvester Dashboard into the Rancher v2.6.4 product results in the creation of VMs that boot to UEFI instead of the OS.
The creation of a VM using _the same settings and configuration_ from the native Harvester dashboard does not result in this behavior. 

**To Reproduce**
1. Go to Virtualization Management w/in Rancher
2. Select Harvester cluster
3. Create a VM, specifying a previously-working image and CPU+mem+network, etc.
4. Observe the VM boots into UEFI instead of the OS

**Expected behavior**
I expected the VM to boot into the OS installed on the image

**Support bundle**

**Environment:**
Harvester v1.0.0
Rancher v2.6.4

**Additional context**
This is an issue because of the presence of these items in the VM manifest:
```
>         features:
>           acpi: {}
>           smm:
>             enabled: false
>         firmware:
>           bootloader:
>             efi:
>               secureBoot: false
```
These are inserted by the experimental Harvester UI in Rancher v2.6.4.

**Comment by WuJun2016** [comment]
Hi, @ebauman   I am trying to reproduce the problem。
1.   What is the image you are using?
2.  Do you expect it to boot using the BIOS?  
3.  Are there any problems if you start UEFI?

workaround:
If you must use BIOS to boot, you can delete the following fields before saving。
```
features:
    acpi: {}
    smm:
      enabled: false
firmware:
  bootloader:
    efi:
      secureBoot: false
```

**Comment by tjjh89017** [comment]
1. un-select `Booting in EFI mode`
UI need to remove
`features.smm`
and remove
`firmware`

```diff
features:
    acpi:
      enabled: true
-    smm:
-      enabled: false
-firmware:
-  bootloader:
-    efi:
-      secureBoot: false
```

2. select `Booting in EFI mode` and `secureBoot: false`

```diff
features:
    acpi:
      enabled: true
    smm:
      enabled: false
firmware:
  bootloader:
    efi:
      secureBoot: false
```

3. select `Booting in EFI mode` and `secureBoot: true`

```diff
features:
    acpi:
      enabled: true
    smm:
-      enabled: false
+      enabled: true
firmware:
  bootloader:
    efi:
-      secureBoot: false
+      secureBoot: true
```

cc @WuJun2016

**Comment by ebauman** [comment]
Even with "Booting in EFI mode" unchecked in the UI, the VM still boots to the UEFI prompt...

**Comment by tjjh89017** [comment]
> Even with "Booting in EFI mode" unchecked in the UI, the VM still boots to the UEFI prompt...

Yes, we understand that, we are working on it to fix it!
thanks for this issue!

**Comment by TachunLin** [comment]
Verified **not fixed** on Harvester `master-03ee6213-head` (04/15) with Rancher `v2.6.4`. 
Move this issue to the implement pipeline. 

### Result
With the default `Booting in EFI mode` unchecked, create a new virtual machine still boots to the UEFI prompt.
  ![image](https://user-images.githubusercontent.com/29251855/163766153-14d8b58b-b3b4-4d05-8d14-dc452965d980.png)
  ![image](https://user-images.githubusercontent.com/29251855/163765993-b2102b93-8976-4a31-a2a7-fbb97990cbf6.png)
  ![image](https://user-images.githubusercontent.com/29251855/163766029-0de95f07-9bf4-4130-a815-799dd8a2992c.png)


### Test Information
* Test Environment: 1 nodes harvester on local kvm machine
* Harvester version: `master-03ee6213-head` (04/15)
* Rancher version: v2.6.4
* OS image: ubuntu focal image

### Verify Steps
1. Create a VM with default setting on Harvester dashboard
2. Import harvester from Rancher v2.6.4 virtualization management
3. Log out Rancher and login again with admin
4. Access Harvester from Rancher virtualization management
5. Create an virtual machine
6. Uncheck  the `Booting in EFI mode` (default) 
7. Open WebVnc while VM power on booting machine
8. Check the screen

### Support Bundle
[supportbundle_28479d0f-c787-4ec6-b759-673c63ce6c76_2022-04-18T06-48-16Z.zip](https://github.com/harvester/harvester/files/8504075/supportbundle_28479d0f-c787-4ec6-b759-673c63ce6c76_2022-04-18T06-48-16Z.zip)

**Comment by TachunLin** [comment]
Verified **`fixed`** on Harvester master-03ee6213-head (04/15) with Rancher v2.6.4. Close this issue

### Result 
After updating the `ui-dashboard-index` url and set the `ui-offline-preferred` to external 

We can create the following types of VM in Harvester and boot correctly  
1. BIOS mode
2. UEFI mode
3. UEFI mode with secure boot

* Uncheck UEFI option

https://user-images.githubusercontent.com/29251855/163794851-18753581-d06c-4298-ba95-d9ab017111ae.mov


* Check UEFI without secure boot option

https://user-images.githubusercontent.com/29251855/163794870-48786f8c-44e8-4f08-a59f-05af99458c24.mov

* Check UEFI with secure boot option

https://user-images.githubusercontent.com/29251855/163795224-4a4f9776-3e67-421d-bfb1-781b61e4edd3.mov


### Verify Steps
1. Create a VM with default setting on Harvester dashboard
2. Import harvester from Rancher v2.6.4 virtualization management
3. Open Global settings
4. Update the ui-dashboard-index value to `https://releases.rancher.com/harvester-ui/dashboard/latest/index.html`
  ![image](https://user-images.githubusercontent.com/29251855/163783590-7c081ef2-b9e8-4ed2-81b1-b12ad74b98c9.png)
5. Change the `ui-offline-preferred` to external 
  ![image](https://user-images.githubusercontent.com/29251855/163785711-f84aa6dd-0592-4535-aa3d-0e6945cf7fa0.png)
6. Log out Rancher and login again with admin
7. Access Harvester from Rancher virtualization management
8. Create an virtual machine
9. Uncheck  the `Booting in EFI mode` (default) 
10. Open WebVnc while VM power on booting machine
11. Check the screen
12. Create another virtual machine
13. Check  the `Booting in EFI mode`
14. Repeat step 10 - 11
15. Create another virtual machine
16. Check  the `Booting in EFI mode` and check the `secure boot` option
17. Repeat step 10 - 11
